libs/pika/async_mpi/src/mpi_polling.cpp

```diff
@@ -26,14 +26,18 @@
 #include <vector>
 
 namespace pika::mpi::experimental {
+
+    constexpr std::uint32_t max_mpi_streams = 4;
```

Comment:
Could this instead be the last member of the `stream_index` enum so that you don't have to keep this in sync with the enum?